dependency/nomad_secure_variables_list_test.go

```diff
@@ -0,0 +1,300 @@
+		i    string
+		exp  *SVListQuery
+		err  bool
+	}{
+		{
+			"empty",
+			"",
+			&SVListQuery{},
+			false,
+		},
+		{
+			"prefix",
+			"prefix",
+			&SVListQuery{
+				prefix: "prefix",
+			},
+			false,
+		},
+		{
+			"dots",
+			"prefix.with.dots",
+			&SVListQuery{
+				prefix: "prefix.with.dots",
+			},
+			false,
+		},
+		{
+			"slashes",
+			"prefix/with/slashes",
+			&SVListQuery{
+				prefix: "prefix/with/slashes",
+			},
+			false,
+		},
+		{
+			"dashes",
+			"prefix-with-dashes",
+			&SVListQuery{
+				prefix: "prefix-with-dashes",
+			},
+			false,
+		},
+		{
+			"leading_slash",
+			"/leading/slash",
+			&SVListQuery{
+				prefix: "leading/slash",
+			},
+			false,
+		},
+		{
+			"trailing_slash",
+			"trailing/slash/",
+			&SVListQuery{
+				prefix: "trailing/slash/",
+			},
+			false,
+		},
+		{
+			"underscores",
+			"prefix_with_underscores",
+			&SVListQuery{
+				prefix: "prefix_with_underscores",
+			},
+			false,
+		},
+		{
+			"special_characters",
+			"config/facet:größe-lf-si",
+			&SVListQuery{
+				prefix: "config/facet:größe-lf-si",
+			},
+			false,
+		},
+		{
+			"splat",
+			"config/*/timeouts/",
+			&SVListQuery{
+				prefix: "config/*/timeouts/",
+			},
+			false,
+		},
+		{
+			"slash",
+			"/",
+			&SVListQuery{
+				prefix: "/",
+			},
+			false,
+		},
+		{
+			"slash-slash",
+			"//",
+			&SVListQuery{
+				prefix: "/",
+			},
+			false,
+		},
+	}
+
+	for i, tc := range cases {
+		t.Run(fmt.Sprintf("%d_%s", i, tc.name), func(t *testing.T) {
+			act, err := NewSVListQuery(tc.i)
+			if (err != nil) != tc.err {
+				t.Fatal(err)
+			}
+
+			if act != nil {
+				act.stopCh = nil
+			}
+
+			assert.Equal(t, tc.exp, act)
+		})
+	}
+}
+
+func TestSVListQuery_Fetch(t *testing.T) {
+
+	type svmap map[string]string
+	_ = testNomad.CreateSecureVariable("test-kv-list/prefix/foo", svmap{"bar": "barp"})
+	_ = testNomad.CreateSecureVariable("test-kv-list/prefix/zip", svmap{"zap": "zapp"})
+	_ = testNomad.CreateSecureVariable("test-kv-list/prefix/wave/ocean", svmap{"sleek": "sleekp"})
+
+	cases := []struct {
+		name string
+		i    string
+		exp  []*NomadSVMeta
+	}{
+		{
+			"exists",
+			"test-kv-list/prefix",
+			[]*NomadSVMeta{
+				{Namespace: "default", Path: "test-kv-list/prefix/foo"},
+				{Namespace: "default", Path: "test-kv-list/prefix/wave/ocean"},
+				{Namespace: "default", Path: "test-kv-list/prefix/zip"},
+			},
+		},
+		{
+			"trailing",
+			"test-kv-list/prefix/",
+			[]*NomadSVMeta{
+				{Namespace: "default", Path: "test-kv-list/prefix/foo"},
+				{Namespace: "default", Path: "test-kv-list/prefix/wave/ocean"},
+				{Namespace: "default", Path: "test-kv-list/prefix/zip"},
+			},
+		},
+		{
+			"no_exist",
+			"test-kv-list/not/a/real/prefix/like/ever",
+			[]*NomadSVMeta{},
+		},
+	}
+
+	for i, tc := range cases {
+		t.Run(fmt.Sprintf("%d_%s", i, tc.name), func(t *testing.T) {
+			d, err := NewSVListQuery(tc.i)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			act, _, err := d.Fetch(testClients, nil)
```

Comment:
It doesn't look like any of the `Fetch` calls include a namespace, so I suspect we don't have good test coverage of the non-default namespace code paths.

Reply:
Currently, the secure variables functions are only configured for the current namespace.  As you mentioned synchronously, there is a need for namespace support for service discovery as well.

Reply:
Updated to support namespaces and region